### PR TITLE
Optimize the acceptance tests testing infrastructure 

### PR DESCRIPTION
**Pull request by HariBhandari07**
# Pull Request Template
This PR optimizes the acceptance tests testing infrastructure of the previous PR https://github.com/jobtriage/jobtriage/pull/99
## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Unit Test
- [ ] Integration Test

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: frontend/tests/steps_file.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+// in this file you can append custom step methods to 'I' object
```
Comment (individual-it) [inline frontend/tests/steps_file.js:1]: what is this file for? it looks empty

File: frontend/tests/acceptance/step_definitions/loginContext.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+const dashboard = require('../pages/dashboardPage');
+const signupPage = require('../pages/signupPage');
+
+Given('the user has browsed to the homepage', () => I.amOnPage('/'));
+
+When('the user browses to the signup page using the webUI', () => I.amOnPage(signupPage.url));
+
+When('the user signs up with name {string}, email {string} password {string} and confirmation password {string} using the webUI',
+  (name, email, password, confirmationPassword) => {
+    signupPage.signUp(name, email, password, confirmationPassword);
```
Comment (individual-it) [inline frontend/tests/acceptance/step_definitions/loginContext.js:12]: is it possible to do that with API calls?

File: frontend/tests/acceptance/features/webUILogin/login.feature [hunk 3]
```diff
@@ -0,0 +1,11 @@
+Feature: login a user
+  As a user
+  I want to login to my dashboard
+
+  Scenario: signup and login a user
+    Given the user has browsed to the homepage
```
Comment (individual-it) [inline frontend/tests/acceptance/features/webUILogin/login.feature:6]: Why browsing to the homepage first when directly after that we browse again to an other page?

File: frontend/codecept.conf.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    I: './tests/steps_file.js',
+    Page: './tests/acceptance/pages/*.js',
+  },
+  mocha: {},
+  bootstrap: null,
+  teardown: null,
+  hooks: [],
+  gherkin: {
+    features: './tests/acceptance/features/*/*.feature',
+    steps: ['./tests/acceptance/step_definitions/loginContext.js'],
```
Comment (kiranparajuli589) [inline frontend/codecept.conf.js:22]:
```suggestion
    steps: ['./tests/acceptance/step_definitions/*.js'],
```
There could be more contexts files in future!

Comment (HariBhandari07) [reply frontend/codecept.conf.js:22]: This is an array, so we can't do that. I tried but it didn't work.

File: frontend/tests/acceptance/features/webUILogin/login.feature [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+    When the user browses to the signup page using the webUI
+    And the user signs up with name "user1", email "user1@gmail.com" password "password" and confirmation password "password" using the webUI
+    And the user browses to the login page
+    And the user logs in with email "user2@gmail.com" and password "hari" using the webUI
```
Comment (kiranparajuli589) [inline frontend/tests/acceptance/features/webUILogin/login.feature:10]: User signed up is with email `user1@gmail.com` but you're logging in with different email-password! Is this intended?

File: frontend/tests/acceptance/pages/signupPage.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+  url: '/signup',
+  fields: {
+    name: '//label[contains(text(),"name")]/parent::div/div/input[contains(@class, "MuiInputBase-input")]',
+    email: '//label[contains(text(),"email")]/parent::div/div/input[contains(@class, "MuiInputBase-input")]',
+    password: '//label[contains(text(),"password")]/parent::div/div/input[contains(@class, "MuiInputBase-input")]',
+    confirmPassword: '//label[contains(text(),"confirm password")]/parent::div/div/input[contains(@class, "MuiInputBase-input")]',
+  },
+  signUp(name, email, password, confirmPassword) {
+    I.waitForVisible(this.fields.email);
+    I.fillField(this.fields.name, email);
```
Comment (HariBhandari07) [inline frontend/tests/acceptance/pages/signupPage.js:13]: name

File: frontend/tests/acceptance/step_definitions/loginContext.js (hunk 2 above)
Comment (HariBhandari07) [reply frontend/tests/acceptance/step_definitions/loginContext.js:10]: @individual-it done in `signUpContext.js`

**Pull request closed without merging**